### ISSUE TITLE
Suggestion | Question: optionally hide Background Music Audio Device

### ISSUE THREAD

**Issue by nodeful** [kyleneideck/BackgroundMusic#136]
Would be nice to have an option to hide the "Background Music Audio Device" if the app is not running. 
Do you know if that is possible at all? I think Boom3D managed to do something like this with their driver. 
I tried to implement something similar on my [eqMac](https://github.com/nodeful/eqMac2) app that uses an old SoundFlower based Kext driver. The only way I found was to hide it by default (read-only property) Then attach it to a created aggregate device to access it (that, however, wiped the volume controls as aggregated devices can't have them)

**Comment by kyleneideck** [comment]
Yeah, it would be really nice if aggregate devices could have controls.

I ran into the same problem with kAudioDevicePropertyIsHidden when I added the [Null Device](https://github.com/kyleneideck/BackgroundMusic/blob/master/BGMDriver/BGMDriver/BGM_NullDevice.h). The workaround I used was to remove the Null Device from the list of devices published by [BGM_PlugIn](https://github.com/kyleneideck/BackgroundMusic/blob/b715212cab3d6d713739faed29885d0c0029ef4e/BGMDriver/BGMDriver/BGM_PlugIn.cpp#L181), BGMDriver's top-level Audio Object. Not sure if you'd be able to do the same thing with the older API SoundFlower's driver uses, though.

I haven't tried it yet, but I think setting kAudioDevicePropertyDeviceCanBeDefaultDevice (and maybe kAudioDevicePropertyDeviceCanBeDefaultSystemDevice) might be better in this case. BGMDevice can still occasionally be useful without having BGMApp open.

**Comment by gchilds** [comment]
BGMDevice is super useful without the app: e.g. Create an aggregate device of Background Music and the  microphone, and another Multi-Output Device/Aggregate with Background music and an output device, setting this to the default output then you've got system audio + mic recording from the first aggregate and the second aggregate handles play through.

The only thing missing stopping this being an epic win is aggregate volume control. Maybe that'll be announced in today's keynote. I don't want to seem ungrateful, but it would also be nice if Apple announced a "first party" pass through device like BGMDevice.
